dakota/src/dom.rs

```diff
@@ -90,54 +90,31 @@ pub struct Content {
     pub el: DakotaId,
 }
 
-/// This is a relative value that modifies an element
-/// by a percentage of the size of the available space.
+/// Represents a possibly relative value. This will
+/// either be a f32 scaling value or a constant size
+/// u32.
 #[derive(Debug, PartialEq, PartialOrd, Copy, Clone)]
-pub struct Relative {
-    val: f32,
+pub enum Value {
+    /// This is a relative value that modifies an element
+    /// by a percentage of the size of the available space.
+    Relative(f32),
+    Constant(i32),
 }
 
-impl Relative {
-    pub fn new(v: f32) -> Self {
-        assert!(v >= 0.0 && v <= 1.0);
-        Self { val: v }
-    }
-
-    pub fn scale(&self, val: f32) -> Result<u32> {
-        if !(self.val >= 0.0 && self.val <= 1.0) {
+impl Value {
+    fn scale(current: f32, val: f32) -> Result<u32> {
+        if !(current >= 0.0 && current <= 1.0) {
             return Err(anyhow!(
                 "Element.relativeOffset should use values in the range (0.0, 1.0)"
             ));
         }
-        Ok((val * self.val) as u32)
+        Ok((current * val) as u32)
```

Comment:
I think we can also change this to return an i32 to avoid the casting mess. This is only called by `get_value` anyway.

Reply:
Updated